webui/react/src/components/WorkspaceCreateModal.tsx

```diff
@@ -323,14 +398,38 @@ const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }
           agentUserGroup?: V1AgentUserGroup;
           checkpointStorageConfig?: unknown;
           name: string;
-          clusterNamespacePairs?: Record<string, string>;
+          clusterNamespaceMeta?: Record<string, V1WorkspaceNamespaceMeta>;
         } = {
           name: workspaceName,
         };
-
-        if (values.bindings) {
-          body['clusterNamespacePairs'] = pick(values.bindings, resourceManagers.getOrElse([]));
-        }
+        const resourceQuotaBody: {
+          id: number;
+          clusterQuotaPairs?: Record<string, number>;
+        } = {
+          id: 0,
+        };
+        const clusterNamespaceMeta = Object.keys(values.bindings ?? {}).reduce(
+          (memo: Record<string, V1WorkspaceNamespaceMeta>, name) => {
+            if (values.bindings !== undefined) {
+              const data = values.bindings[name];
+              if (data.autoCreateNamespace) {
+                data.namespace = undefined;
+                memo[name] = data;
+              } else {
+                if (values.resourceQuotas?.[name]) {
+                  delete values.resourceQuotas[name];
```

Comment:
whats `delete` for?

Reply:
we load the resource quota for each namespace to display in the edit workspace modal. If the user changes the namespace name, we want to delete the older resource quota so that we don't send it in the setResourceQuota API end point while submitting